File: SchedulingFront/src/app/models/product.ts
import { ProductType } from './productType';
import { BasePaging } from '../common/models/basePaging';

export class Product {
    productId: number;
    code: string;
    name: string;
    image: string;
    active: boolean;
    productTypeid: number;
    productType: ProductType;
    price: number;
    organizationUnitId: number;
    organizationUnits = new Array<number>();
    employees: Array<number>;
    productPricelist = new Array<ProductPricelist>();
    organizationUnitsString: string;
}

export class ProductPricelist {
    organizationUnitId: number;
    organizationUnitName: string;
    documentDetailId: number;
    price: number;
    documentId: number;
    productId: number;
    name: string;
    constructor(organizationUnitId: number = null, organizationUnitName: string = null) {
        this.organizationUnitId = organizationUnitId;
        this.organizationUnitName = organizationUnitName;
        this.documentDetailId = null;
        this.price = null;
        this.documentId = null;
    }
}

export class ProductPaging extends BasePaging {
    name: string;
    code: string;
    productTypeId: number;
    organizationUnits = new Array<number>();
}

export class EmployeeOrganizationUnitProduct {
    productId: number;
    name: string;
    organizationUnitId: number;
    employeeId: number;
}
export class ProductSelectList {
    products: Array<Product>;
    productsByOrgUnit: Array<EmployeeOrganizationUnitProduct>;
    productPricelist = new Array<ProductPricelist>();

}
export class ProductSelectListInput {
    allOrgUnits: boolean;
    organizationUnits = new Array<number>();
    constructor(organizationUnits: Array<number>, allOrgUnits = false) {
        this.organizationUnits = organizationUnits;
        this.allOrgUnits = allOrgUnits;
    }
}

File: SchedulingAPI/SchedulingAPI/Models/Product.cs
using Entity.Base;
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;

namespace SchedulingAPI.Models
{
    public class Product
    {
        [PrimaryKey]
        public long? ProductId { get; set; }
        public int? ProductTypeId { get; set; }
        public ProductType ProductType { get; set; }
        public string Code { get; set; }
        public string Name { get; set; }
        public string Image { get; set; }
        public bool Active { get; set; }
        public decimal? Price { get; set; }
        public long? OrganizationUnitId { get; set; }
        public List<ProductPricelist> ProductPricelist { get; set; }
        public List<long> OrganizationUnits { get; set; }
        public string OrganizationUnitsString { get; set; }
    }
    public class ProductPaging : BasePaging
    {
        public string Name { get; set; }
        public string Code { get; set; }
        public List<int> OrganizationUnits = new List<int>();
        public List<int> ProductTypes = new List<int>();
    }

    public class ProductPricelist
    {
        public long? ProductId { get; set; }
        public long? DocumentId { get; set; }
        public int? OrganizationUnitId { get; set; }
        public string OrganizationUnitName { get; set; }
        public decimal? Price { get; set; }
        public long? DocumentDetailId { get; set; }
    }

    public class EmployeeOrganizationUnitProduct
    {
        public long ProductId { get; set; }
        public string Name { get; set; }
        public int OrganizationUnitId { get; set; }
        public int EmployeeId { get; set; }
    }

    //public class ProductsDetails
    //{
    //    public long? ProductId { get; set; }
    //    public string Name { get; set; }
    //    public int? OrganizationUnitId { get; set; }
    //    public string EmployeesString
    //    {
    //        set => Employees = value.Split(",").ToList().Select(x => Convert.ToInt64(x.Trim())).ToList();
    //    }
    //    public List<long> Employees { get; set; }
    //}

    public class ProductSelectList
    {
        public List<Product> Products { get; set; }
        public List<ProductPricelist> ProductPricelist { get; set; }
        public List<EmployeeOrganizationUnitProduct> ProductsByOrgUnit { get; set; }
        public ProductSelectList(List<Product> products,  List<ProductPricelist> productPricelist, List<EmployeeOrganizationUnitProduct> productsByOrgUnit)
        {
            Products = products;
            ProductPricelist = productPricelist;
            ProductsByOrgUnit = productsByOrgUnit;
        }
    }

    public class ProductSelectListInput
    {
        public List<int> OrganizationUnits { get; set; } = new List<int>();
        public bool AllOrgUnits { get; set; }
    }
}